src/Neo.VM/ReferenceCounterV2.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// ReferenceCounterV2.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+
+namespace Neo.VM
+{
+    /// <summary>
+    /// Used for reference counting of objects in the VM.
+    /// </summary>
+    public sealed class ReferenceCounterV2 : IReferenceCounter
+    {
+        private readonly List<StackItem> _stack = [];
+
+        /// <summary>
+        /// Gets the count of references.
+        /// </summary>
+        public int Count => _stack.Count;
+
+        /// <summary>
+        /// Adds item to Reference Counter
+        /// </summary>
+        /// <param name="item">The item to add.</param>
+        /// <param name="count">Number of similar entries</param>
+        public void AddStackReference(StackItem item, int count = 1)
+        {
+            for (var x = 0; x < count; x++)
```

Comment:
this actually is not necessary, currently we process his when we create the compound type. For instance, new array will pass the rc and add all its subitems.

Reply:
I know it, but it's hard to explain, there are cases where the items are removed without exists.

- create a map with items without push the map to stack (unit test)
- Push the map to stack
- Add the map to static variables
- RET

maybe is not this case that i remember, but when the stack was clean, I received "item not found" error